draft-ietf-tls-esni.md

```diff
@@ -804,7 +797,30 @@ connection in this case.
 
 # Security Considerations
 
-## Why is cleartext DNS OK? {#cleartext-dns}
+## Security and Privacy Goals
+
+Informally, the primary security and privacy goals of ECH against an active attacker are
+as follows:
+
+1. Use of ECH does not weaken the security properties of TLS without ECH.
+2. TLS connection establishment to a host with a specific ECHConfig and TLS configuration
+is indistinguishable from a connection to any other host with the same ECHConfig and TLS
+configuration. (The set of hosts which share the same ECHConfig and TLS configuration
+is referred to as the anonymity set.)
+
+Client-facing server configuration determines the size of the anonymity set. For example,
+if a client-facing server uses distinct ECHConfig values for each host, then each anonymity set
+has size k = 1. Client-facing servers SHOULD deploy ECH in such a way so as to maximize the size
+of the anonymity set where possible. This means client-facing servers should use the same
+ECHConfig for as many hosts as possible. An attacker can distinguish two hosts that
+have different ECHConfig values based on the ClientEncryptedCH.record_digest value.
+This also means public information in a TLS handshake is also consistent across hosts.
+For example, if a client-facing server services k backend origin hosts, and exactly one of
+those backend origin hosts supports a different TLS configuration, such as a different
+set of cryptographic algorithms than the other (k - 1) hosts, it may be possible to
+identify this single host based on the contents of unencrypted handshake messages.
```

Comment:
Super nitpicky nitpick: we can probably trim the "For example, [...]" sentence down to:

For example, if a client-facing server services many backend origin hosts, only one of which supports some cipher suite, it may be possible to identify that host based on the contents of unencrypted handshake messages.

(It's already giving an example, and being more specific uses fewer words.)

Reply:
Works for me -- thanks!